src/ConfigFileParser.cpp

```diff
@@ -0,0 +1,534 @@
+
+    int long portNum = std::strtol(port.c_str(), NULL, base);
+    return !(portNum <= minPort || portNum > maxPort);
+}
+
+/**
+ * @brief Checks if the port is valid and reads it if that is the case
+ *
+ * For the case that no semicolon is can be found in the listen directive, the function checks
+ * if the defined values is valid ip address. If that is true, the function returns.
+ * If that is not true, the function and checks if it is a valid port and reads it.
+ *
+ */
+void ConfigFileParser::readPort(const std::string& value)
+{
+    size_t colonIndex = value.find(':');
+    size_t semicolonIndex = value.find(';');
+    std::string port;
+
+    if (colonIndex == std::string::npos) {
+        std::string num = value.substr(0, semicolonIndex);
+        if (isIpAddressValid(num))
+            return;
+        if (value.find('.') != std::string::npos)
+            throw std::runtime_error("Invalid ip address");
+        port = num;
+    } else
+        port = value.substr(colonIndex + 1, semicolonIndex - colonIndex - 1);
+
+    if (!isPortValid(port))
+        throw std::runtime_error("Invalid port");
+
+    for (std::map<std::string, unsigned short>::iterator it = m_configFile.servers[m_configFile.serverIndex].listen.begin(); it != m_configFile.servers[m_configFile.serverIndex].listen.end(); it++)
+        if (it->second == 0)
+            it->second = std::atoi(port.c_str());
+}
+
+/**
+ * @brief Reads the root path
+ *
+ * The function checks if the root path is valid and reads it if that is the case.
+ *
+ * It makes sure that the path is valid in the following ways:
+ * 1. There is only one root path
+ *
+ * If at the end of the path is a slash, it removes it.
+ *
+ * This function can be used for the server block and location block
+ *
+ * @param block The block which surounds the directive
+ */
+void ConfigFileParser::readRootPath(int block, const std::string& value)
+{
+    std::string rootPath = value;
+
+    if (rootPath.find_first_of("  \t\n\v\f\r") != std::string::npos)
+        throw std::runtime_error("More than one root path");
+
+    if (rootPath[rootPath.length() - 1] == '/')
+        rootPath = rootPath.substr(0, rootPath.length() - 1);
+
+    if (block == SERVER)
+        m_configFile.servers[m_configFile.serverIndex].root = rootPath;
+    else if (block == LOCATION)
+        m_configFile.servers[m_configFile.serverIndex].locations[m_configFile.servers[m_configFile.serverIndex].locationIndex].root = rootPath;
+}
+
+/**
+ * @brief Reads and checks the value of the directive in the current line of the config file
+ *
+ * @details This function is called when the directive is valid.
+ *          It calls the appropriate function to read the value of the directive.
+ *          It throws an exception if the value is invalid.
+
+ * @param directive Is the the directive which value is being read and checked
+ */
+void ConfigFileParser::readServerDirectiveValue(const std::string& directive, const std::string& value)
```

Comment:
Why? Because annoying? 

Jason Turner has a vid on it: https://www.youtube.com/watch?v=Zq4yYPG7Erc

Reply:
Interesting vid, but still, feels like over the top, especially when you only have two parameters.

Reply:
For me most checks are over the top. Most of the time I think I can't get this wrong. E.g. here: why should I pass the variables in the wrong order? They are named, I know better!

Still these checks exist precisely because these errors still happen. In this case its even from the suite "bugprone".

If this function would accept a
``` 
struct directive {
  std::string key,
  std::string value };
```
it would not be flagged. 

However: it would still have the same problem: what if you swap the values in the struct? We don't have designated initializers yet so `{ .key = key, .value = value }` is not possible for us.

So yeah I don't think we have a standard, which can make use of this check, so if Chris agrees I would also deactivate.

Reply:
I would also vote for deactivation of this check. 

As Gabriel already mentioned, they have different names. That should be enough to mistake one var with the other

Reply:
Can you please add

`-bugprone-easily-swappable-parameters,`

to clang-tidy file, Checks: 

And at the end as comment short description 

`-bugprone-easily-swappable-parameters : not easily implemented with C++98 standard`

If done you can resolve

Reply:
Done